modules/candidate_parameters/jsx/monitoring.js

```diff
@@ -0,0 +1,449 @@
+              onUserInput ={this.setFormData}
+              ref ={monitorID}
+              disabled ={disabledFields[i] || disabled}
+              required ={elRequired[i]}
+            />
+          );
+
+          visits.push(<hr />);
+          i++;
+        }
+      }
+
+      var formattedHistory = [];
+      for (let row in this.state.Data.history) {
+        if (this.state.Data.history.hasOwnProperty(row)) {
+          let line = [];
+          let lineInfo = "";
+          let historyRow = this.state.Data.history[row];
+          let userH = historyRow.entry_staff;
+          let updateDateH = historyRow.data_entry_date;
+          let visitH = historyRow.visit_label;
+          let flaggedH = historyRow.flag;
+          let monitoredH = historyRow.monitored;
+          let monitoredDateH = historyRow.date_monitored;
+          let examinerH = historyRow.monitor_id;
+
+          lineInfo += "[UPDATE by ";
+          lineInfo += userH;
+          lineInfo += " at ";
+          lineInfo += updateDateH;
+          lineInfo += "]";
+
+          line.push(
+            <td>
+              <b>
+                Visit->
+              </b>
+              {visitH}
+            </td>
+          );
+          line.push(
+            <td class="td-hist">
+              <b>
+                Flagged ->
+              </b>
+              {flaggedH}
+            </td>
+          );
+          line.push(
+            <td class="td-hist">
+              <b>
+                Monitored ->
+              </b>
+              {monitoredH}
+            </td>
+          );
+          if (monitoredDateH !== null) {
+            line.push(
+              <td class="td-hist">
+                <b>
+                  Date Monitored ->
+                </b>
+                {monitoredDateH}
+              </td>
+            );
+          }
+
+          if (examinerH !== null) {
+            line.push(
+              <td class="td-hist">
+                <b>
+                  Monitored by ->
+                </b>
+                {this.state.Data.examiners[examinerH]}
+              </td>
+            );
+          }
+
+          formattedHistory.push(
+            <tr>
+              <td style={{padding: "1px", width: "25%"}}>
+                <i>{lineInfo}</i>
+              </td>
+              {line}
+            </tr>
+            );
+        }
+      }
+
+      var alertMessage = "";
+      var alertClass = "alert text-center hide";
+      if (this.state.updateResult) {
+        if (this.state.updateResult === "success") {
+          alertClass = "alert alert-success text-center";
+          alertMessage = "Update Successful!";
+        } else if (this.state.updateResult === "error") {
+          var errorMessage = this.state.errorMessage;
+          alertClass = "alert alert-danger text-center";
+          alertMessage = errorMessage ? errorMessage : "Failed to update!";
+        }
+      }
+
+      return (
+        <div class="row">
```

Comment:
this should be rendered at the bottom of the page, or where ever the submission button is. otherwise, alert msg will not be seen if candidate has > 2 visits